File: 数据结构与算法/代码/c++/线段树.cpp
#include<stdio.h>
#define MAX 1000
void buildTree(int tree[], int arr[], int treeNode, int start, int end) {
    if (start == end) {
        
        tree[treeNode] = arr[start];
    }
    else {
        int mid = (start + end) / 2;
        int left_node = treeNode * 2 + 1;
        int right_node = treeNode * 2 + 2;

        buildTree(tree, arr, left_node, start, mid);
        buildTree(tree, arr, right_node, mid + 1, end);
        tree[treeNode] = tree[left_node] + tree[right_node];
        printf("%d\n", tree[treeNode]);
    }
}
int main() {
    int arr[] = { 1, 3, 5, 7, 9, 11 };
    int tree[MAX] = { };
    buildTree(tree, arr, 0, 0, 5);
    for (int i = 0; i < 15; i++) {
        printf("terr[%d]=%d\n", i,tree[i]);
    }
    printf("%d",tree[100]);
    return 0;
}

File: 数据结构与算法/算法题/代码/js/线段树.js
function buildTree(tree, arr, treeNode, start, end) {
  if (start == end) {
    tree[treeNode] = arr[start]
    return
  } else {
    let mid = (start + end) / 2
    let left_node = treeNode * 2 + 1
    let right_node = treeNode * 2 + 2
    buildTree(tree, arr, left_node, start, mid)
    buildTree(tree, arr, right_node, mid + 1, end)
    tree[treeNode] = tree[left_node] + tree[right_node]
  }
}
function int(){
    
}
let arr = [1, 3, 5, 7, 9, 11]
let tree = [0]
tree.length=1000
buildTree(tree, arr, 0, 0, arr.length - 1)
console.log(tree)
